src/DBFTPlugin/VRF.cs

```diff
@@ -0,0 +1,365 @@
+            var c = new BigInteger(proof[33..49], true, true);
+            var s = new BigInteger(proof[49..], true, true);
+
+            return Tuple.Create(Gamma, c, s);
+        }
+
+        /// <summary>
+        /// Computes the VRF hash output as result of the digest of a ciphersuite-dependent prefix
+        /// concatenated with the gamma point ([VRF-draft-08](https://tools.ietf.org/pdf/draft-irtf-cfrg-vrf-08), section 5.2).
+        /// </summary>
+        /// <param name="gamma"> An `ECPoint` representing the VRF gamma.</param>
+        /// <returns>A vector of octets with the VRF hash output.</returns>
+        public static byte[] GammaToHash(Cryptography.ECC.ECPoint gamma)
+        {
+            // Multiply gamma with cofactor
+            return SHA256.Create().ComputeHash(new byte[] { (byte)CipherSuite.P256_SHA256_TAI, 0x03 }.Concat(gamma.EncodePoint(true)).ToArray());
+        }
+
+        /// <summary>
+        /// Computes the VRF hash output as result of the digest of a ciphersuite-dependent prefix
+        /// concatenated with the gamma point ([VRF-draft-08](https://tools.ietf.org/pdf/draft-irtf-cfrg-vrf-08), section 5.2).
+        /// </summary>
+        /// <param name="proof"> VRF proof, octet string of length ptLen+n+qLen</param>
+        /// <returns>VRF hash output, octet string of length hLen</returns>
+        public static byte[] ProofToHash(byte[] proof)
+        {
+            var (gamma_point, _, _) = DecodeProof(proof);
+            return GammaToHash(gamma_point);
+        }
+
+        /// <summary>
+        /// Generates proof from a secret key and message as specified in the
+        /// [VRF-draft-08](https://tools.ietf.org/pdf/draft-irtf-cfrg-vrf-08) (section 5.1).
+        /// </summary>
+        /// <param name="prikey"> VRF private key.</param>
+        /// <param name="alpha"> A slice representing the message in octets.</param>
+        /// <returns>- VRF proof, octet string of length ptLen+n+qLen</returns>
+        public static byte[] Prove(byte[] prikey, byte[] alpha)
+        {
+            // Step 1: derive public key from secret key
+            // `Y = x * B`
+            if (prikey.Length != 32) throw new FormatException();
+            var pubkey_point = DerivePubkeyPoint(prikey);
+
+            // Step 2: Hash to curve
+            var h_point = HashToTryAndIncrement(pubkey_point, alpha);
+
+            // Step 3: point to string
+            var h_string = h_point.EncodePoint(true);
+
+            // Step 4: Gamma = x * H
+            var gamma_point = h_point * prikey;
+
+            // Step 5: k = ECVRF_nonce_generation(SK, h_string)
+            var k = GenerateNonce(prikey, h_string);
+            var kBytes = k.ToByteArray(true, true);
+
+            // Step 6: c = ECVRF_hash_points(H, Gamma, k*B, k*H)
+            var u_point = DerivePubkeyPoint(k.ToByteArray(true, true));
+            var v_point = h_point * kBytes;
```

Comment:
> Fixed the error, the issue is neo added extra `length` check to the `prikey` @nicolegys

We should append leading zeros to `kBytes` after `var kBytes = k.ToByteArray(true, true)` in `Prove` method, because the operator `*` requires the second param to be 32 bits.

Reply:
@Liaojinghui I think you've forgotten this issue 0.0

Reply:
> @Liaojinghui I think you've forgotten this issue 0.0

My fault TT